compiler/src/dmd/expression.d

```diff
@@ -3288,12 +3288,19 @@ struct ArgumentList
     }
 }
 
+struct ArgumentLabel
+{
+    Identifier name;    // name of the argument
+    Loc loc;            // location of the argument
+ }
+
 /***********************************************************
  */
 extern (C++) final class CallExp : UnaExp
 {
     Expressions* arguments; // function arguments
     Identifiers* names;     // named argument identifiers
+    ArgumentLabel[] argLabels; // named argument labels
```

Comment:
you will need yo update the corresponding C++ header with a `DArray<ArgumentLabel>`, also `ArgumentLabel` will need to be `extern(C++)`. You will also need to update `frontend.h` accordingly.